src/execution_plan/optimizations/optimizations_util.c

```diff
@@ -0,0 +1,59 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "optimizations_util.h"
+
+void re_order_filter_op(ExecutionPlan *plan, OpBase *root, OpFilter *filter) {
+	FT_FilterNode *additional_filter_tree = filter->filterTree;
+
+	rax *references = FilterTree_CollectModified(additional_filter_tree);
```

Comment:
I think a one-liner would be a bit more clear:
`rax *references = FilterTree_CollectModified(filter->filterTree);`

Reply:
changed